apps/mull-api/src/app/user/user.module.ts

```diff
@@ -1,12 +1,11 @@
 import { Module } from '@nestjs/common';
-import { TypeOrmModule } from '@nestjs/typeorm';
 import { UserResolver } from './user.resolver';
 import { UserService } from './user.service';
 
-import { User } from '../entity/user.entity';
+import { EntitiesModule } from '../entities';
 
 @Module({
-  imports: [TypeOrmModule.forFeature([User])],
+  imports: [EntitiesModule],
```

Comment:
I saw this got merged while I was reviewing still but I'll still ask this for clarification.

So I'm not familiar with Angular best practices but if I'm correct here user.module imports all entities instead of specifically what it needs. Is this how it should be done? @osahtout @TheGreatMarkus

Reply:
As far as I know, if a module import another module, it just makes whatever the imported module has to offer available for importing, it doesn't import it all up-front.

Reply:
I don't think it will affect imports or bloat things up more, its mostly just a way to organize code. If you need access to the entities, you need to import the entities module, and then you'll have access to what it offers.

Reply:
That would make sense, thanks!

Reply:
I saw it in @ghiobi's branch, maybe he can explain why. 🤔

Reply:
The entities module encapsulates everything from the entities folder. From the previous codebase, we were defining the User entity in two places, the root module, and usermodule. The entities module is the only place that will declare the entities and your service modules such as usermodule will need this module to inject the repository.

https://docs.nestjs.com/techniques/database#auto-load-entities

Maybe in the future, you can move the entities folder into a library `@mull/entities` and then have your separate libraries for `users`, `media` and etc. so that you can develop microservices.